lasagne/layers/recurrent.py

```diff
@@ -0,0 +1,1347 @@
+
+        # Create single recurrent computation step function
+        def step(input_n, hid_previous, *args):
+            # Compute the hidden-to-hidden activation
+            hid_pre = helper.get_output(self.hidden_to_hidden, hid_previous)
+
+            # If the dot product is precomputed then add it, otherwise
+            # calculate the input_to_hidden values and add them
+            if self.precompute_input:
+                hid_pre += input_n
+            else:
+                hid_pre += helper.get_output(self.input_to_hidden, input_n)
+
+            # Clip gradients
+            if self.grad_clipping is not False:
+                hid_pre = theano.gradient.grad_clip(
+                    hid_pre, -self.grad_clipping, self.grad_clipping)
+
+            return self.nonlinearity(hid_pre)
+
+        def step_masked(input_n, mask_n, hid_previous, *args):
+            # Skip over any input with mask 0 by copying the previous
+            # hidden state; proceed normally for any input with mask 1.
+            hid = step(input_n, hid_previous, *args)
+            hid_out = hid*mask_n + hid_previous*(1 - mask_n)
+            return [hid_out]
+
+        if mask is not None:
+            mask = mask.dimshuffle(1, 0, 'x')
+            sequences = [input, mask]
+            step_fun = step_masked
+        else:
+            sequences = input
+            step_fun = step
+
+        # When hid_init is provided as a TensorVariable, use it as-is
+        if isinstance(self.hid_init, T.TensorVariable):
+            hid_init = self.hid_init
+        else:
+            # Dot against a 1s vector to repeat to shape (num_batch, num_units)
+            hid_init = T.dot(T.ones((num_batch, 1)), self.hid_init)
+
+        if self.unroll_scan:
+            # Retrieve the dimensionality of the incoming layer
+            input_shape = self.input_shapes[0]
+            # Explicitly unroll the recurrence instead of using scan
+            hid_out = unroll_scan(
+                fn=step_fun,
+                sequences=sequences,
+                outputs_info=[hid_init],
+                go_backwards=self.backwards,
+                non_sequences=non_seqs,
+                n_steps=input_shape[1])[0]
+        else:
+            # Scan op iterates over first dimension of input and repeatedly
+            # applies the step function
+            hid_out = theano.scan(
+                fn=step_fun,
+                sequences=sequences,
+                go_backwards=self.backwards,
+                outputs_info=[hid_init],
+                non_sequences=non_seqs,
+                truncate_gradient=self.gradient_steps,
+                strict=True)[0]
+
+        # dimshuffle back to (n_batch, n_time_steps, n_features))
+        hid_out = hid_out.dimshuffle(1, 0, 2)
+
+        # if scan is backward reverse the output
+        if self.backwards:
+            hid_out = hid_out[:, ::-1, :]
+
+        self.hid_out = hid_out
+        return hid_out
```

Comment:
I think that should be removed. I guess i added it because I had not figured out that Lasagne supported a regular way to get that output.
There is a similar problem in line 992 with the LSTM cell. I don't think you can refer to the cell in other ways than with the through the class attribute ?